src/algebra/group_power/basic.lean

```diff
@@ -387,23 +346,69 @@ alias add_sq ← add_pow_two
 
 end comm_semiring
 
-@[simp] lemma neg_sq {α} [ring α] (z : α) : (-z)^2 = z^2 :=
+section ring
+variable [ring R]
+
+section
+variables (R)
```

Comment:
```suggestion
variables (R) [ring R]

```

Reply:
I'd prefer to leave this the way I have it, for two reasons:
* This is how it was already done in this file, I just moved it
* Your way is much more likely to cause new lemmas to be added with explicit `R` arguments accidentally.

Is that convincing enough for you, or do you still want me to change it?

Reply:
Ooh, I really don't care. The pattern I'm suggesting is quite common in mathlib, and it is a bit shorter. But I see the advantages of your style as well. So feel free to merge it as is.